userapi/src/test/java/fk/prof/userapi/api/MockAggregatedProfile.java

```diff
@@ -0,0 +1,419 @@
+                .setZone("chennai-1")
+                .setInstanceType("c1.xlarge").build();
+
+        FinalizedProfileWorkInfo wi = new FinalizedProfileWorkInfo(1, r1, AggregationState.COMPLETED, aggregationStart.plusSeconds(10), aggregationStart.plusSeconds(10 + 60), 60,
+                buildMap("full-app-trace", 5),
+                buildMap(AggregatedProfileModel.WorkType.cpu_sample_work, count1)
+                );
+
+        FinalizedProfileWorkInfo wi2 = new FinalizedProfileWorkInfo(1, r2, AggregationState.RETRIED, aggregationStart.plusSeconds(24), aggregationStart.plusSeconds(24 + 60), 60,
+                buildMap("full-app-trace", 10),
+                buildMap(AggregatedProfileModel.WorkType.cpu_sample_work, count2)
+                );
+
+        return buildMap(101l, wi, 102l, wi2);
+    }
+
+    // A temp byteBuffer is reused everytime to write a abstractMessage. This avoids the creation of temp buffer everytime
+    // you want to write the message but its content is copied to the outputStream.
+    public void testCodedOutputStream_firstWriteToByteBufferThenCopyToOutputStream() throws Exception {
+        ByteBuffer b = ByteBuffer.allocate(1000);
+
+        ByteArrayOutputStream bos = new ByteArrayOutputStream(20_000_000);
+        CodedOutputStream finalout = CodedOutputStream.newInstance(bos);
+
+        long milli = System.currentTimeMillis();
+        int count = 0;
+        int limit = 1_000_000;
+
+        for(count = 0; count < limit; count++) {
+            AggregatedProfileModel.Header h = getHeader();
+
+            CodedOutputStream cos = CodedOutputStream.newInstance(b);
+            h.writeTo(cos);
+            cos.flush();
+
+            int size = b.position();
+            b.flip();
+
+            finalout.writeUInt32NoTag(size);
+            finalout.write(b);
+            finalout.flush(); //flush to array outstream
+
+            b.clear();
+
+            if(bos.size() > 10_000_000) {
+                bos.reset();
+            }
+        }
+
+        long milli2 = System.currentTimeMillis();
+
+        System.out.println("Time took: " + (milli2 - milli));
+    }
+
+    // here message is directly written to outputStream but a temp buffer is always created in the process.
+    public void testCodedOutputStream_directlyWriteToOutputStreamWUsingWriteDelimited() throws Exception {
+
+        ByteArrayOutputStream bos = new ByteArrayOutputStream(20_000_000);
+
+        long milli = System.currentTimeMillis();
+        int count = 0;
+        int limit = 1_000_000;
+
+        for(count = 0; count < limit; count++) {
+            AggregatedProfileModel.Header h = getHeader();
+
+            h.writeDelimitedTo(bos);
+
+            if(bos.size() > 10_000_000) {
+                bos.reset();
+            }
+        }
+
+        long milli2 = System.currentTimeMillis();
+
+        System.out.println("Time took: " + (milli2 - milli));
+    }
+
+    private AggregatedProfileModel.Header getHeader() {
```

Comment:
Not required here if the above two methods are going to be moved to a separate class